### ISSUE TITLE
error compile extfat-nofuse for new kernel 4.14.*

### ISSUE THREAD

**Issue by build000** [openwrt/packages#5444]
@br101 @champtar @hnyman or other
machine: WNDR4700
target: apm821xx
version:
```
scripts/getver.sh
r5812-d667909
```
Compile:
```
make -j8 BUILD_LOG=1
```
Log:
```
cat logs/package/feeds/packages/exfat-nofuse/compile.txt
```
```
make[4]: Entering directory '/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/linux-4.14.13'
  CC [M]  /home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_core.o
  CC [M]  /home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.o
/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.c: In function 'exfat_create':
/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.c:733:7: error: 'CURRENT_TIME_SEC' undeclared (first use in this function)
  ts = CURRENT_TIME_SEC;
       ^
/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.c:733:7: note: each undeclared identifier is reported only once for each function it appears in
/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.c: In function 'exfat_unlink':
/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.c:889:7: error: 'CURRENT_TIME_SEC' undeclared (first use in this function)
  ts = CURRENT_TIME_SEC;
       ^
/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.c: In function 'exfat_symlink':
/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.c:934:7: error: 'CURRENT_TIME_SEC' undeclared (first use in this function)
  ts = CURRENT_TIME_SEC;
       ^
/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.c: In function 'exfat_mkdir':
/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.c:1012:7: error: 'CURRENT_TIME_SEC' undeclared (first use in this function)
  ts = CURRENT_TIME_SEC;
       ^
/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.c: In function 'exfat_rmdir':
/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.c:1067:7: error: 'CURRENT_TIME_SEC' undeclared (first use in this function)
  ts = CURRENT_TIME_SEC;
       ^
/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.c: In function 'exfat_rename':
/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.c:1131:7: error: 'CURRENT_TIME_SEC' undeclared (first use in this function)
  ts = CURRENT_TIME_SEC;
       ^
/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.c: In function 'exfat_cont_expand':
/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.c:1205:36: error: 'CURRENT_TIME_SEC' undeclared (first use in this function)
  inode->i_ctime = inode->i_mtime = CURRENT_TIME_SEC;
                                    ^
/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.c: At top level:
/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.c:1388:19: error: initialization from incompatible pointer type [-Werror=incompatible-pointer-types]
  .getattr       = exfat_getattr,
                   ^
/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.c:1388:19: note: (near initialization for 'exfat_dir_inode_operations.getattr')
/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.c:1422:17: error: 'generic_readlink' undeclared here (not in a function)
  .readlink    = generic_readlink,
                 ^
/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.c: In function '_exfat_truncate':
/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.c:1490:36: error: 'CURRENT_TIME_SEC' undeclared (first use in this function)
  inode->i_ctime = inode->i_mtime = CURRENT_TIME_SEC;
                                    ^
/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.c: At top level:
/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.c:1514:17: error: initialization from incompatible pointer type [-Werror=incompatible-pointer-types]
  .getattr     = exfat_getattr,
                 ^
/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.c:1514:17: note: (near initialization for 'exfat_file_inode_operations.getattr')
/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.c: In function 'exfat_write_end':
/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.c:1686:37: error: 'CURRENT_TIME_SEC' undeclared (first use in this function)
   inode->i_mtime = inode->i_ctime = CURRENT_TIME_SEC;
                                     ^
/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.c: In function 'exfat_read_root':
/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.c:2416:7: error: 'CURRENT_TIME_SEC' undeclared (first use in this function)
  ts = CURRENT_TIME_SEC;
       ^
cc1: some warnings being treated as errors
scripts/Makefile.build:314: recipe for target '/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.o' failed
make[5]: *** [/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/exfat_super.o] Error 1
Makefile:1505: recipe for target '_module_/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3' failed
make[4]: *** [_module_/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3] Error 2
make[4]: Leaving directory '/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/linux-4.14.13'
Makefile:52: recipe for target '/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/.built' failed
make[3]: *** [/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700_FAST_PATH/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/exfat-nofuse-2017-01-03-8d291f525ce6d88fe0d8b11b86fd5c2e900401d3/.built] Error 2
```

**Comment by bouwew** [comment]
Just FYI: https://github.com/openwrt/packages/pull/5388

**Comment by build000** [comment]
@bouwew - confirmed - this patch https://github.com/openwrt/packages/pull/5388/commits/fb9c77d2310b1d38ad097030dfa34fdc2c210242 (PR #5388) to extfat-nofuse fixing compilation and build kernel module properly (fs-extfat):

cat logs/package/feeds/packages/extfat-nofuse/compile.txt
```
+ curl -f --connect-timeout 20 --retry 5 --location --insecure https://codeload.github.com/dorimanx/exfat-nofuse/tar.gz/de4c760?dummy=/exfat-nofuse-snapshot-20170619.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 73088    0 73088    0     0  66917      0 --:--:--  0:00:01 --:--:-- 66930
make[4]: Entering directory '/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700-TRUNK/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/linux-4.14.13'
  CC [M]  /home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700-TRUNK/build_dir/target-powerpc_464fp_musl/exfat-nofuse-de4c760/exfat_core.o
  CC [M]  /home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700-TRUNK/build_dir/target-powerpc_464fp_musl/exfat-nofuse-de4c760/exfat_super.o
  CC [M]  /home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700-TRUNK/build_dir/target-powerpc_464fp_musl/exfat-nofuse-de4c760/exfat_api.o
  CC [M]  /home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700-TRUNK/build_dir/target-powerpc_464fp_musl/exfat-nofuse-de4c760/exfat_blkdev.o
  CC [M]  /home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700-TRUNK/build_dir/target-powerpc_464fp_musl/exfat-nofuse-de4c760/exfat_cache.o
  CC [M]  /home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700-TRUNK/build_dir/target-powerpc_464fp_musl/exfat-nofuse-de4c760/exfat_data.o
  CC [M]  /home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700-TRUNK/build_dir/target-powerpc_464fp_musl/exfat-nofuse-de4c760/exfat_bitmap.o
  CC [M]  /home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700-TRUNK/build_dir/target-powerpc_464fp_musl/exfat-nofuse-de4c760/exfat_nls.o
  CC [M]  /home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700-TRUNK/build_dir/target-powerpc_464fp_musl/exfat-nofuse-de4c760/exfat_oal.o
  CC [M]  /home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700-TRUNK/build_dir/target-powerpc_464fp_musl/exfat-nofuse-de4c760/exfat_upcase.o
  LD [M]  /home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700-TRUNK/build_dir/target-powerpc_464fp_musl/exfat-nofuse-de4c760/exfat.o
  Building modules, stage 2.
  MODPOST 1 modules
  CC      /home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700-TRUNK/build_dir/target-powerpc_464fp_musl/exfat-nofuse-de4c760/exfat.mod.o
  LD [M]  /home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700-TRUNK/build_dir/target-powerpc_464fp_musl/exfat-nofuse-de4c760/exfat.ko
make[4]: Leaving directory '/home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700-TRUNK/build_dir/target-powerpc_464fp_musl/linux-apm821xx_nand/linux-4.14.13'
rstrip.sh: /home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700-TRUNK/build_dir/target-powerpc_464fp_musl/exfat-nofuse-de4c760/ipkg-powerpc_464fp/kmod-fs-exfat/lib/modules/4.14.13/exfat.ko: relocatable
Packaged contents of /home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700-TRUNK/build_dir/target-powerpc_464fp_musl/exfat-nofuse-de4c760/ipkg-powerpc_464fp/kmod-fs-exfat into /home/sj/Pulpit/Projekty/LEDE-REBOOT_FAST_PATH/WNDR4700-TRUNK/bin/targets/apm821xx/nand/packages/kmod-fs-exfat_4.14.13+snapshot-20170619-1_powerpc_464fp.ipk
```
Unfortunately, I can not comment on the operation of this module because currently I do not have any media with this file system.

**Comment by yousong** [comment]
This should now be fixed by #5494
